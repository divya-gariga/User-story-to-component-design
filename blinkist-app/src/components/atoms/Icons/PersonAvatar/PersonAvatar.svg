<svg width="14" height="19" viewBox="0 0 14 19" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M13.6667 18.3333H12V16.6666C12 16.0036 11.7366 15.3677 11.2678 14.8989C10.799 14.43 10.1631 14.1666 9.50004 14.1666H4.50004C3.837 14.1666 3.20111 14.43 2.73227 14.8989C2.26343 15.3677 2.00004 16.0036 2.00004 16.6666V18.3333H0.333374V16.6666C0.333374 15.5616 0.772361 14.5018 1.55376 13.7204C2.33516 12.939 3.39497 12.5 4.50004 12.5H9.50004C10.6051 12.5 11.6649 12.939 12.4463 13.7204C13.2277 14.5018 13.6667 15.5616 13.6667 16.6666V18.3333ZM7.00004 10.8333C6.34343 10.8333 5.69325 10.704 5.08662 10.4527C4.48 10.2014 3.9288 9.83314 3.46451 9.36885C3.00021 8.90455 2.63192 8.35336 2.38064 7.74673C2.12937 7.1401 2.00004 6.48992 2.00004 5.83331C2.00004 5.1767 2.12937 4.52652 2.38064 3.9199C2.63192 3.31327 3.00021 2.76207 3.46451 2.29778C3.9288 1.83349 4.48 1.46519 5.08662 1.21392C5.69325 0.962642 6.34343 0.833313 7.00004 0.833313C8.32612 0.833313 9.59789 1.3601 10.5356 2.29778C11.4733 3.23546 12 4.50723 12 5.83331C12 7.1594 11.4733 8.43116 10.5356 9.36885C9.59789 10.3065 8.32612 10.8333 7.00004 10.8333ZM7.00004 9.16665C7.8841 9.16665 8.73194 8.81546 9.35706 8.19034C9.98218 7.56521 10.3334 6.71737 10.3334 5.83331C10.3334 4.94926 9.98218 4.10141 9.35706 3.47629C8.73194 2.85117 7.8841 2.49998 7.00004 2.49998C6.11598 2.49998 5.26814 2.85117 4.64302 3.47629C4.0179 4.10141 3.66671 4.94926 3.66671 5.83331C3.66671 6.71737 4.0179 7.56521 4.64302 8.19034C5.26814 8.81546 6.11598 9.16665 7.00004 9.16665Z" fill="#6D787E"/>
</svg>
