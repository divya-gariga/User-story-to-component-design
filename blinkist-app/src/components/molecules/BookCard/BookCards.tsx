import React from "react";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import "./card.css";
import { Time } from "../../atoms/Icons/icons/icons";

interface CardProps {
  img: string;
  title: string;
  author: string;
  readCount: string;
  readTime: string;
  increasedHeight: string;
  personAvatar: string;
}
const BookCards = (props: CardProps) => {

  return (
    <Card
      sx={{
        maxWidth: "282px",
        borderBottom: "1px solid white",
        borderRadius: "8px 8px 0px 0px",
        height: "max-Content",

      }}
    >
      <CardMedia
        component="img"
        image={props.img}

        height={props.increasedHeight} alt="media"
        sx={{ "objectFit": 'contain' }}
      />
      <CardContent>
        <div className="book-card_title">{props.title}</div>
        <div className="book-card_author">{props.author}</div>
        <div className="book-card_stats">
          <div className="book-card_stats-item">
            <span className="icon">
              <Time />
            </span>
            <span className="book-card_readTime" > 13-minutes read</span>
            <span className="book-card_reads">
              <span className="book-card_icon"><img src={props.personAvatar} /></span>
              <span className="book-card_readCount">{props.readCount}</span>
            </span>
          </div>
        </div>
        <div className="ellipsis"><b>...</b></div>
      </CardContent>
    </Card >
  );
};
export default BookCards;
