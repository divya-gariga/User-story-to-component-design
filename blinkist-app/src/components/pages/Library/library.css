.tab-on-hover:hover {
  color: yellow;
}
.tabs.Mui-selected {
  color: "#00FF00"
}
.tabs.Mui-focused {
  color: "#00FF00"
}