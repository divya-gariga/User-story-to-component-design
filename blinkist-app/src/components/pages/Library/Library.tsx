import { Grid, Typography } from "@mui/material";
import React, { useEffect, useState } from "react";
import Box from "@mui/material/Box";
import Tab from "@mui/material/Tab";
import TabContext from "@mui/lab/TabContext";
import TabList from "@mui/lab/TabList";
import TabPanel from "@mui/lab/TabPanel";
import "./library.css";
import BookCards from "../../molecules/BookCard/BookCards";
import axios from "axios";
import Primary from "../../molecules/Buttons/Primary/Primary";
import LineProgressBar from "../../atoms/Icons/ProgressBar/LineProgressBar";
import PersonAvatar from "../../atoms/Icons/PersonAvatar/PersonAvatar.svg";
import { Link } from "react-router-dom";

interface BookProps {
  id: number;
  coverImage: string;
  title: string;
  author: string;
  state: string;
  isTrending: boolean;
  isJustAdded: boolean;
  isFeatured: boolean;
  personAvatar: string;
  readCount: string;
}
function Library() {
  let displayAvatar: any;
  const [value, setValue] = React.useState("1");
  const [blinksInfo, setBlinksInfo] = useState([]);
  const [blinkState, setBlinkState] = useState(false);
  const handleChange = (_event: React.SyntheticEvent, newValue: string) => {
    setValue(newValue);
  };
  useEffect(() => {
    axios
      .get("http://localhost:8000/BooksInfo")
      .then((response) => {
        const data = response.data;
        setBlinksInfo(data);
      })
      .catch((error) => console.error("Error :", error));
  }, [blinkState]);
  function onClickOfFinished(book: BookProps, id: number) {
    axios
      .put(`http://localhost:8000/BooksInfo/${id}`, {
        ...book,
        state: "finished",
      })
      .then((response) => console.log(response))
      .catch((error) => {
        console.log(error);
      });
    setBlinkState(!blinkState);
  }
  function onClickOfReadAgain(book: BookProps, id: number) {
    axios
      .put(`http://localhost:8000/BooksInfo/${id}`, {
        ...book,
        state: "reading",
      })
      .then((response) => console.log(response))
      .catch((error) => {
        console.log(error);
      });
    setBlinkState(!blinkState);
  }
  function cardBuild(book: BookProps, progressVal: number, label: string,) {
    if (book["readCount"] === "") {
      displayAvatar = null;
    }
    else {
      displayAvatar = PersonAvatar;
    }
    let button;
    if (label === "Finished") {
      button = <Primary
        label={label}
        displayElement={{ display: "" }}
        buttonBorderRadius={{ borderRadius: "0px" }}
        onClickHandler={() =>
          onClickOfFinished(book, book["id"])
        } data-testid="finishedButtton"
      />
    } else {
      button =
        <Primary
          label={label}
          displayElement={{ display: "" }}
          buttonBorderRadius={{
            borderRadius: "0px",
          }}
          onClickHandler={() =>
            onClickOfReadAgain(book, book["id"])
          } data-testid="readAgainButton"
        />
    }
    return (
      <Grid item md={4} pb={"25px"} key={book["id"]} data-testid={`$book["id"]`} >
        <Box width={"300px"} data-testid={`$book["id"]`}>
          <Link to={`/Entrepreneurship/${book["id"]}`} style={{ textDecoration: "none" }}>

            <BookCards
              img={book["coverImage"]}
              title={book["title"]}
              author={book["author"]}
              readTime={"13-minutes read"}
              increasedHeight="310px"
              readCount={book["readCount"]}
              personAvatar={displayAvatar}
            />
          </Link>

          {button}
          <LineProgressBar
            displayElement={{ display: "" }}
            progressValue={progressVal}
            topHeight={{ top: "0px" }} data-testid="bar"
          />
        </Box>
      </Grid>
    );
  }
  const styles = {
    heading: {
      fontFamily: "CeraProBold",
      fontSize: "36px",
      lineHeight: "45px",
      fontWeight: 700,
    },
    sample: {
      fontFamily: 'CeraProMedium'

    },
    link: {
      textDecoration: "none",
      color: "#6D787E",
      fontFamily: "Cera Pro",
      borderBottom: "2px solid grey",
      paddingRight: "10%",
    },
    tabs: {
      "&.MuiButtonBase-root.MuiTab-root": {
        backgroundColor: "yellow",
      },
    },
  };
  return (
    <>
      <div style={{ padding: "100px 17.97% 111px 18.33%" }}>
        <Typography style={styles.heading} pb={"48px"}>
          My Library
        </Typography>
        <Box sx={{ width: "100%" }} data-testid="box">
          <TabContext value={value} >
            <Box sx={{ borderBottom: 1, borderColor: "divider" }}>
              <span style={{ color: "#22C870x" }}>
                <TabList
                  onChange={handleChange}
                  textColor={"secondary"}
                  TabIndicatorProps={{
                    style: {
                      backgroundColor: "rgba(44, 224, 128, 1)",
                      width: "304px"
                    },
                  }}
                >
                  <Tab
                    label="Currently Reading"
                    value="1"
                    sx={{
                      textTransform: "capitalize",
                      fontFamily: "CeraProLight",
                      fontWeight: "700",
                      fontSize: "18px",
                      width: "304px",
                      alignItems: "flex-start",
                      paddingLeft: "0px"
                    }}
                  />
                  <Tab
                    label="Finished"
                    value="2"
                    sx={{
                      textTransform: "capitalize",
                      fontFamily: "CeraProLight",
                      fontWeight: "700",
                      fontSize: "18px",
                      alignItems: "flex-start",
                      paddingLeft: "0px"
                    }}
                  />
                </TabList>
              </span>
            </Box>
            <TabPanel value="1" sx={{ paddingLeft: "0px", paddingRight: "0px" }} data-testid="panel1">
              <Grid container columnSpacing={"4%"} data-testid="contatiner1">
                {blinksInfo
                  .filter((book) => {
                    if (book["readCount"] === "") {
                      displayAvatar = null;
                    }
                    else {
                      displayAvatar = PersonAvatar;
                    }
                    return book["state"] === "reading";
                  })
                  .map((book) => cardBuild(book, 35, "Finished"))
                }
              </Grid>
            </TabPanel>
            <TabPanel value="2" sx={{ paddingLeft: "0px", paddingRight: "0px" }} data-testid="panel2">
              <Grid container columnSpacing={"4%"} data-testid="contatiner2">
                {blinksInfo
                  .filter((book) => {
                    return book["state"] === "finished";
                  })
                  .map((book) => cardBuild(book, 100, "Read Again"))}
              </Grid>
            </TabPanel>
          </TabContext>
        </Box>
      </div>
    </>
  );
}
export default Library;