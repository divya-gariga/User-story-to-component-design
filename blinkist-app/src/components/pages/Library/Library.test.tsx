import React from 'react';
import { render, screen } from '@testing-library/react';
import Library from './Library';
import renderer from 'react-test-renderer';
import PersonAvatar from "../../components/atoms/Icons/PersonAvatar/PersonAvatar.svg";
import BookCards from '../../molecules/BookCard/BookCards';
import axios from 'axios';
import update from './Library'

test('typography', () => {
    render(<Library />)
    expect(screen.getByText(/My Library/i)).toBeInTheDocument();
    expect(screen.getByText(/Currently Reading/i)).toBeInTheDocument();
    expect(screen.getByText(/Finished/i)).toBeInTheDocument();
    expect(screen.getByTestId("panel1")).toBeInTheDocument();
    expect(screen.getByTestId("panel1")).toBeInTheDocument();
    // expect(screen.getByTestId("bar")).toBeInTheDocument();
    // expect(screen.getByTestId("1")).toBeInTheDocument();
    expect(screen.getByTestId("box")).toBeInTheDocument();
    // expect(screen.getByTestId("container1")).toBeInTheDocument();
    // /expect(screen.getByTestId("container2")).toBeInTheDocument();
    // expect(screen.getByTestId("finishedButton")).toBeInTheDocument();
    // expect(screen.getByTestId("readAgainButton")).toBeInTheDocument();

})

test("snapshot for app component 1", () => {
    const renderedComponent = renderer.create(<Library />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
})

test("bookcard",()=>{
    render(  <BookCards
        img={"assets/beingboss.png"}
        increasedHeight="310px"
        title={"Being Boss"}
        author={"Kathleen Shannon and Emily"}
        readTime={"13-minutes read"}
        readCount={"16k reads"}
        personAvatar={PersonAvatar}
      />)
      expect(screen.getByText(/Being Boss/i)).toBeInTheDocument();
      expect(screen.getByText(/Kathleen Shannon and Emily/i)).toBeInTheDocument();
      expect(screen.getByText(/13-minutes read/i)).toBeInTheDocument();
      expect(screen.getByText(/16k reads/i)).toBeInTheDocument();
      expect(screen.getByAltText(/media/i)).toBeInTheDocument();
    })
// jest.mock('axios');
// const mockedAxios = axios as jest.Mocked<typeof axios>;

// describe("fetch data",()=>{
//     it("fetch successfully ",async ()=>{
//         const data={
//             data:{
//             BooksInfo: [
//                 {
//                   "id": 1,
//                   "coverImage": "assets/beingboss.png",
//                   "title": "Being Boss",
//                   "author": "Kathleen Shannon and Emily",
//                   "state": "finished",
//                   "isTrending": true,
//                   "isJustAdded": false,
//                   "isFeatured": false,
//                   "link": "#",
//                   "readCount": ""
//                 }]}
//         };
//         mockedAxios.get.mockImplementationOnce(()=>Promise.resolve(data))
//         render(<Library/>);
//         await expect().resolves.toEqual(data);
//     });
  
// })