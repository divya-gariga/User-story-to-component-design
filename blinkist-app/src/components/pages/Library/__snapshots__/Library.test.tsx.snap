// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for app component 1 1`] = `
<div
  style={
    Object {
      "padding": "50px 17.97% 111px 18.33%",
    }
  }
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1fbtkm8-MuiTypography-root"
    style={
      Object {
        "fontFamily": "CeraProBold",
        "fontSize": "36px",
        "fontWeight": 700,
        "lineHeight": "45px",
      }
    }
  >
    My Library
  </p>
  <div
    className="MuiBox-root css-8atqhb"
    data-testid="box"
  >
    <div
      className="MuiBox-root css-1gsv261"
    >
      <span
        style={
          Object {
            "color": "#22C870x",
          }
        }
      >
        <div
          className="MuiTabs-root css-1ujnqem-MuiTabs-root"
        >
          <div
            className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            onScroll={[Function]}
            style={
              Object {
                "marginBottom": -0,
                "overflow": "hidden",
              }
            }
          >
            <div
              aria-orientation={null}
              className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              onKeyDown={[Function]}
              role="tablist"
            >
              <button
                aria-controls={null}
                aria-selected={true}
                className="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary Mui-selected css-11ivv5d-MuiButtonBase-root-MuiTab-root"
                disabled={false}
                id={null}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex={0}
                type="button"
              >
                Currently Reading
                <span
                  className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                  style={
                    Object {
                      "backgroundColor": "rgba(44, 224, 128, 1)",
                      "width": "304px",
                    }
                  }
                />
              </button>
              <button
                aria-controls={null}
                aria-selected={false}
                className="MuiButtonBase-root MuiTab-root MuiTab-textColorSecondary css-74vjky-MuiButtonBase-root-MuiTab-root"
                disabled={false}
                id={null}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="tab"
                tabIndex={-1}
                type="button"
              >
                Finished
              </button>
            </div>
          </div>
        </div>
      </span>
    </div>
    <div
      aria-labelledby={null}
      className="MuiTabPanel-root css-jigkhi-MuiTabPanel-root"
      data-testid="panel1"
      hidden={false}
      id={null}
      role="tabpanel"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-1ofkode-MuiGrid-root"
        data-testid="contatiner1"
      />
    </div>
    <div
      aria-labelledby={null}
      className="MuiTabPanel-root css-jigkhi-MuiTabPanel-root"
      data-testid="panel2"
      hidden={true}
      id={null}
      role="tabpanel"
    />
  </div>
</div>
`;
